src/test/librbd/fsx.cc

```diff
@@ -1183,7 +1185,30 @@ krbd_resize(struct rbd_ctx *ctx, uint64_t size)
 	if (ret < 0)
 		return ret;
 
-	return __librbd_resize(ctx, size);
+	ret = __librbd_resize(ctx, size);
+	if (ret < 0)
+		return ret;
+
+	while (count++ < 10) {
+		ret = krbd_get_size(ctx, &effective_size);
+		if (ret < 0) {
+			prt("krbd_get_size failed\n");
+			return ret;
+		}
+
+		ret = 0;
+		if (size != effective_size) {
+			prt("Sizes: expected (0x%llx), effective (0x%llx)\n",
+			    (unsigned long long)size,
+			    (unsigned long long)effective_size);
+			sleep(count);
+			ret = -EINVAL;
+			continue;
+		}
+		break;
```

Comment:
Using all of `break`, `continue` and provisionally setting `ret` to handle running out of iterations makes this hard to follow.  If you want to stick to `count`, let's do something like:
```
	for (;;) {
		ret = krbd_get_size(ctx, &effective_size);
		if (ret < 0)
			return ret;

		if (effective_size == size)
			break;

		if (count++ >= 15) {
			prt("...");
			return -EINVAL;
		}

		usleep(count * 250 * 1000);
	}
```